### ISSUE TITLE
Pauses of almost a second after alternate selection

### ISSUE THREAD

**Issue by calavera** [apache/trafficserver#6342]
We're noticing that there are some unusual pauses after alternate selection happens. These pauses are around 1 second per alternate in the selection vector. We're still trying to find the cause of problem, but I thought I'd open an issue to see if someone else has seen this behavior before.

We're running 7.1.x, and we're about to try 9.0.x to see if the behavior is also there. You can see in the timestamps below how the continuation seems to stop after the alternate selection for almost a minute, and then HttpCacheSM sends an OPEN_READ_FAILED event:

```
[Jan 23 04:44:19.104] Server {0x7f074deb2700} DEBUG: <HttpTransactCache.cc:265 (SelectFromAlternates)> (http_seq) [SelectFromAlternates] Chosen alternate # 1
[alts] and the winner is alternate number 1
[Jan 23 04:44:20.043] Server {0x7f074deb2700} DEBUG: <HttpCacheSM.cc:118 (state_cache_open_read)> (http_cache) [40] [&HttpCacheSM::state_cache_open_read, CACHE_EVENT_OPEN_READ_FAILED]
```

We're seen the problem in isolated environments with only healthchecks sending request to TrafficServer and us sending two or three requests with curl. It happens with and without compression.

We also have the three settings below in our records configuration, which might be part of the problem:

```
# retries cache 5 times adding up to a total of 500ms delay into non cached requests.
CONFIG proxy.config.http.cache.max_open_read_retries INT 5
CONFIG proxy.config.http.cache.open_read_retry_time INT 100
# enables cache read while another request to the same url is writing to the same entry.
CONFIG proxy.config.cache.enable_read_while_writer INT 1
```

Does anyone have any idea about what might be happening?

**Comment by calavera** [comment]
Still early to confirm for sure, but it looks like 9.0.x doesn't have this problem. In the graph below, we stopped seeing this behavior at 14:15 when we deployed 9.0.x to one of our production nodes:

![Screenshot from 2020-01-23 15-43-07](https://user-images.githubusercontent.com/1050/73033549-398d4f00-3df7-11ea-96d2-03f35858f9c1.png)

Something worth mentioning that I think I didn't say before is that all the transactions with the issue were marked as CACHE_MISS, even thought we know for sure that the object was in the cache, but I guess that's because of the OPEN_READ_FAILED event.

**Comment by calavera** [comment]
Fixed in 7.1.9 released last Friday.
